README.md

```diff
@@ -1,189 +1,785 @@
+
+
+        override fun onCreate(savedInstanceState: Bundle?) {
+            super.onCreate(savedInstanceState)
+            setContentView(R.layout.activity_main)
+
+            signInButton.isEnabled = false
+
+            val appDomain = URI("https://flamboyant-darwin-d11c17.netlify.com")
+            val redirectURI = URI("${appDomain}/redirect")
+            val manifestURI = URI("${appDomain}/manifest.json")
+            val scopes = arrayOf(Scope.StoreWrite)
+
+            _blockstackSession = BlockstackSession(this, appDomain, redirectURI, manifestURI, scopes,
+                    onLoadedCallback = {signInButton.isEnabled = true
+                    })
+
+
+            signInButton.setOnClickListener { view: View ->
+                blockstackSession().redirectUserToSignIn { userData ->
+                    runOnUiThread {
+                        onSignIn(userData)
+                    }
+                }
+            }
+            if (intent?.action == Intent.ACTION_VIEW) {
+                handleAuthResponse(intent)
+            }
+        }
+
+        private fun onSignIn(userData: UserData) {
+            userDataTextView.text = "Signed in as ${userData.did}"
+
+            signInButton.isEnabled = false
+
+        }
+
+        override fun onNewIntent(intent: Intent?) {
+            super.onNewIntent(intent)
+
+            if (intent?.action == Intent.ACTION_MAIN) {
+                blockstackSession().loadUserData {
+                    userData -> runOnUiThread {if (userData != null) {onSignIn(userData)}}
+                }
+            } else if (intent?.action == Intent.ACTION_VIEW) {
+                handleAuthResponse(intent)
+            }
+        }
+
+        private fun handleAuthResponse(intent: Intent) {
+            val response = intent.dataString
+            if (response != null) {
+                val authResponseTokens = response.split(':')
+
+                if (authResponseTokens.size > 1) {
+                    val authResponse = authResponseTokens[1]
+
+                    blockstackSession().handlePendingSignIn(authResponse, { userData ->
+                        // The user is now signed in!
+                        runOnUiThread {
+                            onSignIn(userData)
+                        }
+                    })
+                }
+            }
+        }
+
+        fun blockstackSession() : BlockstackSession {
+            val session = _blockstackSession
+            if(session != null) {
+                return session
+            } else {
+                throw IllegalStateException("No session.")
+            }
+        }
+
+
+    }
+    ```
+
+### Run the final app in the emulator
+
+1. Choose **Run > Apply changes**.
+2. Choose **Run > Run app** in the emulator.
+3. When you see the application open, choose **Sign in with Blockstack**.
+
+    The system prompts you how to open.
+
+    ![](images/chrome-prompt.png)
+
+4. Choose **Chrome** and click **ALWAYS**.
```

Comment:
It also works with Firefox :-)

Reply:
Thanks @friedger. Are you seeing Firefox as an option in your emulator? I'm surprised Google allows that to come up in their emulator.